### PR TITLE
Bumped vault to 0.6.1

### PR DESCRIPTION
**Pull request by geofffranks**

### CODE REVIEW

**Comment by jhunt** [comment]
There was some concern with changes to the way HA Vaults operated when moving from 0.6.0 to 0.6.1.  @norman-abramovitz would be able to provide more details.

**Comment by geofffranks** [comment]
Ok. We will need to hold off on app role for concourse integrations until we get to 0.6.1 or beyond.

**Comment by norman-abramovitz** [comment]
1.  The concern was making sure the HA upgrade process upgrades all the standby nodes before upgrading the active node.   Also all nodes must be shut down cleanly as documented in the 0.6.1 upgrade document. https://github.com/hashicorp/vault/blob/master/website/source/docs/install/upgrade-to-0.6.1.html.md   We have not done a Vault bosh release upgrade yet.
2. WIth only HA 3 nodes there is a window when vault could be not protected in HA mode until the last node is upgraded iff a failure occurs on the new active node.   We should warn the customer.
3.  Some new health check status codes have been added that need to be tested out if they will cause any issue.
4. There is a new endpoint for creating a root certificate(generate-root) which might need to be incorporated into safe.  The directions for codex would at least to be changed to use this new endpoint.
5. Should we figure out how to switch from AppID to AppRole now?

The bottom line we should be diligent in testing HA mode in this release upgrade.   It is not a simple replace the executable and go like before.
